src/lksearch/catalog.py

```diff
@@ -0,0 +1,399 @@
+"""Functions to search various catalogs for missions"""
+
+from typing import Union
+import numpy as np
+from astropy.coordinates import Angle, SkyCoord, Distance
+from astropy.table import Table
+from astropy.time import Time
+import astropy.units as u
+from astroquery.vizier import Vizier
+import pandas as pd
+
+import warnings
+
+__all__ = ["query_KIC", "query_EPIC", "query_TIC", "query_gaia", "query_catalog"]
+
+# This is a lits of VizieR catalogs and their input parameters to be used in the
+# query_skycatalog function
+_Catalog_Dictionary = {
+    "kic": {
+        "catalog": "V/133/kic",
+        "columns": ["KIC", "RAJ2000", "DEJ2000", "pmRA", "pmDE", "Plx", "kepmag"],
+        "column_filters": "kepmag",
+        "rename_in": ("KIC", "pmDE", "kepmag"),
+        "rename_out": ("ID", "pmDEC", "Kepler_Mag"),
+        "equinox": Time(2000, format="jyear", scale="tt"),
+        "prefix": "KIC",
+    },
+    "epic": {
+        "catalog": "IV/34/epic",
+        "columns": ["ID", "RAJ2000", "DEJ2000", "pmRA", "pmDEC", "plx", "Kpmag"],
+        "column_filters": "Kpmag",
+        "rename_in": ("Kpmag", "plx"),
+        "rename_out": ("K2_Mag", "Plx"),
+        "equinox": Time(2000, format="jyear", scale="tt"),
+        "prefix": "EPIC",
+    },
+    "tic": {
+        "catalog": "IV/39/tic82",
+        "columns": ["TIC", "RAJ2000", "DEJ2000", "pmRA", "pmDE", "Plx", "Tmag"],
+        "column_filters": "Tmag",
+        "rename_in": ("TIC", "pmDE", "Tmag"),
+        "rename_out": ("ID", "pmDEC", "TESS_Mag"),
+        "equinox": Time(2000, format="jyear", scale="tt"),
+        "prefix": "TIC",
+    },
+    "gaiadr3": {
+        "catalog": "I/355/gaiadr3",
+        "columns": ["DR3Name", "RAJ2000", "DEJ2000", "pmRA", "pmDE", "Plx", "Gmag"],
+        "column_filters": "Gmag",
+        "rename_in": ("DR3Name", "pmDE", "Gmag"),
+        "rename_out": ("ID", "pmDEC", "Gaia_G_Mag"),
+        "equinox": Time(2016, format="jyear", scale="tt"),
+        "prefix": None,
+    },
+}
+
+
+def _apply_propermotion(table: Table, equinox: Time, epoch: Time):
+    """
+    Returns an astropy table of sources with the proper motion correction applied
+
+    Parameters:
+    -----------
+    table :
+        astropy.table.Table which contains the coordinates of targets and proper motion values
+    equinox: astropy.time.Time
+        The equinox for the catalog
+    epoch : astropy.time.Time
```

Comment:
How will this be called in the workflow? Catalog values won't have a relevant epoch, so is this expected to be called from a lksearch result for actual data?

Reply:
As this is passed from epoch in the query calls, I would be more explicit in the documentation that this time is used for PM corrections. I would also suggest this is optional.  Right now I think how 'epoch' is set in this is confusing.

Reply:
I think users need this functionality, but I've set it to default to the catalogs equinox.